library/core/src/unicode/printable.py

```diff
@@ -211,7 +211,11 @@ def main():
 pub(crate) fn is_printable(x: char) -> bool {
     let x = x as u32;
     let lower = x as u16;
-    if x < 0x10000 {
+
+    if lower >= 32 && lower < 127 {
+        // ASCII fast path
+        true
+    } else if x < 0x10000 {
```

Comment:
I don't think this is quite correct. `lower` is just the lower 16 bits, so this conditional would also match any character whose lower 16 bits is in 32..=126, even if the upper bits are non-zero. And as far as I can tell, I don't think that's actually true for all of Unicode; for instance, `U+E0020` isn't printable.

Reply:
Good catch, that is definitely wrong